python_modules/dagster/dagster/_config/pythonic_config/resource.py

```diff
@@ -834,7 +834,7 @@ def _is_annotated_as_resource_type(annotation: type, metadata: list[str]) -> boo
     """Determines if a field in a structured config class is annotated as a resource type or not."""
     from dagster._config.pythonic_config.type_check_utils import safe_is_subclass
 
-    if metadata and metadata[0] == "resource_dependency":
+    if metadata and any(m == "resource_dependency" for m in metadata):
```

Comment:
union mode `left_to_right` ends up in this metadata list, bumping resource dependency to second